src/main/java/com/desiato/puresynth/services/AudioService.java

```diff
@@ -1,51 +1,58 @@
 package com.desiato.puresynth.services;
 
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 import javax.sound.sampled.*;
 import java.io.ByteArrayInputStream;
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 
+@Slf4j
 @Service
 public class AudioService {
 
-    private static final int SAMPLE_RATE = 44100;
-    private static final int BITS_DEPTH = 16;
-    private static final boolean BIG_ENDIAN = false;
+    private final FileStorageService fileStorageService;
 
-    public byte[] generateSineWaveFile(
-            double frequency,
-            double durationInSeconds)
-            throws IOException {
+    @Autowired
+    public AudioService(FileStorageService fileStorageService) {
+        this.fileStorageService = fileStorageService;
+    }
 
-        // Generate the sine wave data as a byte array
+    public byte[] generateSineWaveFile(double frequency, double durationInSeconds) throws IOException {
         byte[] buffer = generateSineWaveBuffer(frequency, durationInSeconds);
 
-        // Create an audio format
-        AudioFormat format = new AudioFormat(SAMPLE_RATE, BITS_DEPTH, 1, true, BIG_ENDIAN);
+        AudioFormat format = new AudioFormat(
+                44100,
+                16,
+                1,
+                true,
+                false);
 
-        // Use ByteArrayOutputStream to avoid saving to disk
         try (ByteArrayInputStream bais = new ByteArrayInputStream(buffer);
              ByteArrayOutputStream baos = new ByteArrayOutputStream();
              AudioInputStream ais = new AudioInputStream(bais, format, buffer.length / 2)) {
+
             AudioSystem.write(ais, AudioFileFormat.Type.WAVE, baos);
             return baos.toByteArray();
         }
     }
 
-    private byte[] generateSineWaveBuffer(double frequency, double durationInSeconds) {
-        int bufferLength = (int) (durationInSeconds * SAMPLE_RATE);
+    public String saveAudioFile(byte[] audioBytes, String fileName) throws IOException {
+        // Save the file using the FileStorageService (local or cloud storage)
```

Comment:
The comment just duplicates info from the method name and body. It's obvious it saves a file